src/offlinelogstorage/dlt_offline_logstorage_behavior.c

```diff
@@ -543,9 +598,13 @@
 
         ret = stat(absolute_file_path, &s);
 
-        /* if size is enough, open it */
-        if ((ret == 0) && (s.st_size + msg_size <= (int)config->file_size)) {
-            dlt_logstorage_open_log_output_file(config, absolute_file_path, "a");
+        /* if file stats is read and, either
+         * is_sync is true and (other than ON_MSG sync behavior and current size is less than configured size) or
+         * msg_size fit into the size (ON_MSG or par of cache needs to be written into new file), open it */
+        if ((ret == 0) &&
+            ((is_sync && (s.st_size < (int) config->file_size)) ||
+             (!is_sync && (s.st_size + msg_size <= (int) config->file_size)))) {
+            config->log = fopen(absolute_file_path, "a+");
```

Comment:
Hello @michael-methner 
This is a typical check result from CodeQL, please note this for a reference about how codeQL work.
Regards

Reply:
Hello @minminlittleshrimp , hello @lvklevankhanh ,
i think this race condition can be avoided by using fopen first and perform the check with fstat() instead of stat(). If size is to small, file should be immediately closed and error handling performed.